web/src/components/SecondaryNav.test.js

```diff
@@ -1,9 +1,81 @@
 import { shallow } from 'enzyme';
 import React from 'react';
-import SecondaryNav from './SecondaryNav';
+import {
+  plain as SecondaryNav,
+  mapStateToProps,
+  mapDispatchToProps
+} from './SecondaryNav';
+import { addActivity } from '../actions/editActivity';
+import { jumpTo } from '../actions/app';
+
+global.scrollTo = jest.fn();
 
 describe('Secondary Nav component', () => {
-  it('renders correctly', () => {
-    expect(shallow(<SecondaryNav />).dive()).toMatchSnapshot();
+  const props = {
+    activities: [{ anchor: 'activity-1' }, { anchor: 'activity-2' }],
+    activityCount: 2,
+    add: jest.fn(),
+    jumpTo: jest.fn()
+  };
+
+  beforeEach(() => {
+    props.add.mockClear();
+    props.jumpTo.mockClear();
+  });
+
+  it('renders correctly without add activity button', () => {
+    expect(
+      shallow(<SecondaryNav {...props} activeSection="activity-1-ffp" />).dive()
+    ).toMatchSnapshot();
+  });
+
+  it('renders correctly with add activity button', () => {
+    expect(
+      shallow(<SecondaryNav {...props} activeSection="activity-2-ffp" />).dive()
+    ).toMatchSnapshot();
+  });
+
+  it('handles add activity button click', () => {
+    const component = shallow(
+      <SecondaryNav {...props} activeSection="activity-2-ffp" />
+    ).dive();
+    const link = component.find('.ds-c-button');
+    expect(link.props().to).toBe('/apd/activities');
+
+    link.simulate('click');
+
+    expect(props.add).toHaveBeenCalled();
+    expect(props.jumpTo).toHaveBeenCalledWith('activities-list');
+    expect(window.scrollTo).toHaveBeenCalledWith(0, 0);
+  });
+
+  it('maps state to props', () => {
+    const state = {
+      apd: {
+        data: {
+          activities: [
+            { anchor: 'activity-1', name: 'activity', key: '1' },
+            { anchor: 'activity-2', name: 'activity', key: '2' }
```

Comment:
```suggestion
            { name: 'activity', key: '1' },
            { name: 'activity', key: '2' }
```

The activities in the app state don't have an `anchor` property, so I think it's best if they also don't have them in a test fixture. The `anchor` property that gets returned by the selector should be computed.